lib/ansible/modules/monitoring/zabbix/zabbix_host_facts.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) me@mimiko.me
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+RETURN = '''
+---
+hosts:
+  description: List of Zabbix host. See https://www.zabbix.com/documentation/3.4/manual/api/reference/host/get for list of host values.
+  returned: success
+  type: dict
+  sample: [ { "available": "1", "description": "", "disable_until": "0", "error": "", "flags": "0", "groups": ["1"], "host": "Host A", ... } ]
+'''
+
+DOCUMENTATION = '''
+---
+module: zabbix_host_facts
+short_description: Gather facts about Zabbix host
+description:
+   - This module allows you to search for Zabbix host entries.
+version_added: "2.6"
+author:
+    - "(@redwhitemiko)"
+requirements:
+    - "python >= 2.6"
+    - zabbix-api
+options:
+    server_url:
+        description:
+            - Url of Zabbix server, with protocol (http or https).
+        required: true
+        aliases: [ "url" ]
+    login_user:
+        description:
+            - Zabbix user name, used to authenticate against the server.
+        required: true
+    login_password:
+        description:
+            - Zabbix user password.
+        required: true
+    http_login_user:
+        description:
+            - Basic Auth login.
+        required: false
+        default: null
+    http_login_password:
+        description:
+            - Basic Auth password.
+        required: false
+        default: null
+    host_name:
+        description:
+            - Name of the host in Zabbix.
+            - host_name is the unique identifier used and cannot be updated using this module.
+        required: true
+    host_ip:
+        description:
+            - Host interface IP of the host in Zabbix.
+        required: false
+    timeout:
+        description:
+            - The timeout of API request (seconds).
+        default: 10
+    exact_match:
+        description:
+            - Find the exact match
+        type: bool
+        default: False
+    remove_duplicate:
+        description:
+            - Remove duplicate host from host result
+        type: bool
+        default: False
```

Comment:
not a blocker but it is usual to use `yes/no` instead of `True/False` in the docs for bool types.